packages/js-core/src/website/lib/widget.ts

```diff
@@ -156,27 +136,55 @@ const renderWidget = async (
 
         const { id } = res.data;
 
+        const existingDisplays = websiteConfig.get().personState.data.displays;
+        const displays = existingDisplays ? [...existingDisplays, survey.id] : [survey.id];
+        const previousConfig = websiteConfig.get();
+
+        const updatedPersonState: TJsPersonState = {
+          ...previousConfig.personState,
+          data: {
+            ...previousConfig.personState.data,
+            displays,
+            lastDisplayAt: new Date(),
+          },
+        };
+
+        const filteredSurveys = filterPublicSurveys(previousConfig.environmentState, updatedPersonState);
+
+        websiteConfig.update({
+          ...previousConfig,
+          environmentState: previousConfig.environmentState,
+          personState: updatedPersonState,
+          filteredSurveys,
+        });
+
         surveyState.updateDisplayId(id);
         responseQueue.updateSurveyState(surveyState);
       },
       onResponse: (responseUpdate: TResponseUpdate) => {
-        const displays = websiteConfig.get().state.displays;
+        const displays = websiteConfig.get().personState.data.displays;
         const lastDisplay = displays && displays[displays.length - 1];
         if (!lastDisplay) {
           throw new Error("No lastDisplay found");
         }
-        if (!lastDisplay.responded) {
-          lastDisplay.responded = true;
-          const previousConfig = websiteConfig.get();
-          let state = filterPublicSurveys({
-            ...previousConfig.state,
-            displays,
-          });
-          websiteConfig.update({
-            ...previousConfig,
-            state,
-          });
-        }
+
+        const responses = websiteConfig.get().personState.data.responses;
+        const newPersonState: TJsPersonState = {
+          ...websiteConfig.get().personState,
+          data: {
+            ...websiteConfig.get().personState.data,
+            responses: [...responses, surveyState.surveyId],
+          },
+        };
```

Comment:
Run this on the response creation only, as `onResponse` runs on each answer submission

Reply:
similar to `packages/js-core/src/app/lib/widget.ts#L197`